RiotSharp/Requester.cs

```diff
@@ -19,72 +20,112 @@ internal Requester(string apiKey = "")
         public string CreateGetRequest(string relativeUrl, string rootDomain, List<string> addedArguments = null,
             bool useHttps = true)
         {
-            this.rootDomain = rootDomain;
-            var request = PrepareRequest(relativeUrl, addedArguments, useHttps);
-            return GetResponse(request);
+            RootDomain = rootDomain;
+            var request = PrepareRequest(relativeUrl, addedArguments, useHttps, HttpMethod.Get);
+            return GetResult(request);
         }
 
         public async Task<string> CreateGetRequestAsync(string relativeUrl, string rootDomain,
             List<string> addedArguments = null, bool useHttps = true)
         {
-            this.rootDomain = rootDomain;
-            var request = PrepareRequest(relativeUrl, addedArguments, useHttps);
-            return await GetResponseAsync(request);
+            RootDomain = rootDomain;
+            var request = PrepareRequest(relativeUrl, addedArguments, useHttps, HttpMethod.Get);
+            return await GetResultAsync(request);
         }
 
-        protected HttpWebRequest PrepareRequest(string relativeUrl, List<string> addedArguments, bool useHttps,
-            string httpMethod = "GET")
+        protected HttpRequestMessage PrepareRequest(string relativeUrl, List<string> addedArguments,
+            bool useHttps, HttpMethod httpMethod)
         {
-            HttpWebRequest request;
-            string scheme = useHttps ? "https" : "http";
-            if (addedArguments == null)
+            var scheme = useHttps ? "https" : "http";
+            var url = addedArguments == null ?
+                $"{scheme}://{RootDomain}{relativeUrl}?api_key={ApiKey}" :
+                $"{scheme}://{RootDomain}{relativeUrl}?{BuildArgumentsString(addedArguments)}api_key={ApiKey}";
+
+            return new HttpRequestMessage(httpMethod, url);
+        }
+
+        protected string GetResult(HttpRequestMessage request)
+        {
+            var result = string.Empty;
+            try
             {
-                request = (HttpWebRequest)WebRequest.Create(string.Format("{0}://{1}{2}?api_key={3}"
-                    , scheme, rootDomain, relativeUrl, ApiKey));
+                using (var client = new HttpClient())
```

Comment:
I've always been used to disposing things after any http requests, but according to [this](http://stackoverflow.com/questions/22560971/what-is-the-overhead-of-creating-a-new-httpclient-per-call-in-a-webapi-client) it seems it's meant to be reused.